edx_enterprise_subsidy_client/client.py

```diff
@@ -189,3 +209,61 @@ def can_redeem(self, subsidy_uuid, lms_user_id, content_key):
         )
         response.raise_for_status()
         return response.json()
+
+
+class EnterpriseSubsidyAPIClientV2(EnterpriseSubsidyAPIClient):  # pylint: disable=abstract-method
```

Comment:
should the v2 client really be extending a class containing v1 endpoints?  Like, this would result in carrying over retrieve_subsidy_transaction() from v1, despite the class name having v2 in it, right?  Also it would carry over         `/api/v1/content-metadata/<content_identifier>/` endpoint which is fine, but that's also technically a v1 endpoint.  Seems confusing, but I won't block.